webapp/channels/src/selectors/views/custom_status.test.ts

```diff
@@ -14,7 +14,14 @@ import configureStore from 'store';
 import {TestHelper} from 'utils/test_helper';
 import {addTimeToTimestamp, TimeInformation} from 'utils/utils';
 
-jest.mock('mattermost-redux/selectors/entities/users');
+jest.mock('mattermost-redux/selectors/entities/users', () => {
```

Comment:
This may be due to a cyclic reference, or jest.mock not handling well the way we are exporting things.

The problem is... this test file requires `selectors/entities/users`, which depends on `post_utils.ts`, which depends on `selectors/entities/roles.ts`, which in turn depends on `selectors/entities/users.ts` to access `getCurrentUserId`, which is really defined in `selectors/entities/common.ts`. All this is needed just because we create the selectors on importing the file.

If we don't do this, the test will complain that `getCurrentUserId` is not defined.

Reply:
Yeah, that could be the problem. Cyclic references cause a lot of problems with `createSelector` because imported selectors can be `undefined` at the time the file is loaded due to a circular reference, and then that causes problems because `createSelector` runs at load time